plugins/Kaleidoscope-Colormap-Overlay/src/kaleidoscope/plugin/Colormap-Overlay.h

```diff
@@ -53,6 +53,19 @@ class ColormapOverlay : public kaleidoscope::Plugin {
     overlay_count_ = _overlay_count;
   }
 
+  template<uint8_t _layer_count>
+  void configureOverlays(uint8_t const (&overlays)[_layer_count][kaleidoscope_internal::device.matrix_rows * kaleidoscope_internal::device.matrix_columns]) {
+    overlays_      = Overlay[];
+    overlay_count_ = 0;
+    for (int layer_ = 0; layer_ < _layer_count; layer_++) {
+      for (int key_index_ = 0; key_index_ < kaleidoscope_internal::device.matrix_rows * kaleidoscope_internal::device.matrix_columns; key_index_++) {
+        color_index_ = overlays[layer_][key_index_];
+        // TODO(EvyBongers): validate the color index?
+        overlays_[overlay_count_] = Overlay(layer_, /* TODO(EvyBongers) */, color_index_);
```

Comment:
@obra I haven't been able to find a convenience function to translate a key index to a key address. I'm also unsure if I can construct a `KeyAddr` from the key index. I probably overlooked a constructor for it?

Reply:
KeyAddr is currently always implemented as a MatrixAddr.

The key index is described as `offset_` inside MatrixAddr.

Here's what we've got for constructors in there:

```

  constexpr MatrixAddr(uint8_t row, uint8_t col)
    : offset_(row * cols + col) {}

  explicit constexpr MatrixAddr(uint8_t offset)
    : offset_(offset) {}

```

The actual calculation of an offset looks like this: (from `src//kaleidoscope/device/key_indexes.h`)

```
constexpr uint8_t keyIndex(uint8_t row, uint8_t col) {
  return KeyAddr(row, col).toInt() + 1;
}
```